src/render/mod.rs

```diff
@@ -789,31 +780,36 @@ impl Renderer {
                 }
                 ref other => {
                     let (pso, color, param0, map) = match *other {
-                        Material::MeshPbr { .. } => unreachable!(),
-                        Material::LineBasic { color } => (&self.pso.line_basic, color, 0.0, None),
-                        Material::MeshBasic {
-                            color,
-                            ref map,
-                            wireframe: false,
-                        } => (&self.pso.mesh_basic_fill, color, 0.0, map.as_ref()),
-                        Material::MeshBasic {
-                            color,
-                            map: _,
-                            wireframe: true,
-                        } => (&self.pso.mesh_basic_wireframe, color, 0.0, None),
-                        Material::MeshLambert { color, flat } => (
+                        Params::Pbr(_) => unreachable!(),
+                        Params::Basic(ref params) => {
+                            match params.pipeline {
```

Comment:
tip: Code size can be reduced by moving match inside created tuple

Reply:
This is what `rustfmt` chose, I think. :frowning: